packages/edit-site/src/index.js

```diff
@@ -51,6 +52,26 @@ export function reinitializeEditor( target, settings ) {
 		return;
 	}
 
+	/*
+	 * Prevent adding the Clasic block in the site editor.
+	 * Only add the filter when the site editor is initialized, not imported.
+	 * Also only add the filter(s) after registerCoreBlocks()
+	 * so that common filters in the block library are not overwritten.
+	 *
+	 * This usage here is inspired by previous usage of the filter in the post editor:
+	 * https://github.com/WordPress/gutenberg/pull/37157
+	 */
+	addFilter(
```

Comment:
Did we try just unregistering (or not registering) the block instead of using a filter?

Reply:
I didn't! 

Let me try it out quickly.

Reply:
I see something quite strange: when **not** registering the block, three `core/freeform` blocks get inserted into the site editor: 

https://user-images.githubusercontent.com/5417266/198698173-fe786c34-61e0-4ef3-ad5a-853c4497cc0c.mp4

https://user-images.githubusercontent.com/5417266/198698183-f8225373-16ac-4ab7-9ef8-5120de6f9459.mp4

Any idea why that might be? I now remember that I've seen this behaviour before when working on this PR which is why I've opted for the fix by using a filter but I didn't investigate the root cause.

Reply:
I guess, it's something in the parser (probably related about the fallback block being used). Also makes me wonder what happens now in trunk when you edit the HTML of templates manually and add random stuff. Do we have a block that handles random HTML.

Reply:
In the widget editor there was also a situation where freeform blocks were inserted, but I think that only happened when pasting blocks. I guess it wasn't happening when loading content because the widget editor doesn't store blocks as html, while the site editor does.

The fix that was used in the widget editor was to use the html block as the freeform block (instead of the classic block). But that caused some other issues - [no block delimiter for the html block](https://github.com/WordPress/gutenberg/issues/35266) and [autop running on freeform content](https://github.com/WordPress/gutenberg/pull/42623).

My suggestion for a longer term fix would be a [new (non-classic) freeform block](https://github.com/WordPress/gutenberg/issues/23086#issuecomment-1250722344). It probably wouldn't be too difficult to implement, borrowing parts from the html and classic block.